src/handlers/guilds/GUILD_DELETE.ts

```diff
@@ -1,13 +1,28 @@
 import { eventHandlers } from "../../bot.ts";
+import { cacheHandlers } from "../../cache.ts";
 import { DiscordPayload, GuildDeletePayload } from "../../types/mod.ts";
 import { basicShards } from "../../ws/shard.ts";
-import { cacheHandlers } from "../../cache.ts";
 
 export async function handleGuildDelete(
   data: DiscordPayload,
   shardID: number,
 ) {
   const payload = data.d as GuildDeletePayload;
+
+  const guild = await cacheHandlers.get("guilds", payload.id);
+  if (!guild) return;
+
+  await cacheHandlers.delete("guilds", payload.id);
+
+  if (payload.unavailable) {
+    const shard = basicShards.get(shardID);
+    if (shard) shard.unavailableGuildIDs.add(payload.id);
```

Comment:
why here?

Reply:
What exactly do you mean?

Reply:
Because of this in guild create 
```ts
  if (shard?.unavailableGuildIDs.has(payload.id)) {
    await cacheHandlers.delete("unavailableGuilds", payload.id);

    shard.unavailableGuildIDs.delete(payload.id);
  }
```

Reply:
Alrigth why is that in guildCreate?

Reply:
Because of this in READY.ts
```ts
  // All guilds are unavailable at first
  shard.unavailableGuildIDs = new Set(payload.guilds.map((g) => g.id));
```